extensions/mailer/deployment/src/test/java/io/quarkus/mailer/FakeSmtpServer.java

```diff
@@ -0,0 +1,183 @@
+package io.quarkus.mailer;
+
+import java.util.Locale;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.NetServerOptions;
+import io.vertx.core.parsetools.RecordParser;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.core.net.NetServer;
+import io.vertx.mutiny.core.net.NetSocket;
+
+public class FakeSmtpServer {
+
+    private NetServer netServer;
+    private String[][] dialogue;
```

Comment:
I think these should be volatile, as they are set from the test thread but read from the IO thread.